src/pki.rs

```diff
@@ -199,8 +200,91 @@ fn verify_signature(
+        //     verified = verifying_key.verify(message, &signature).is_ok();
+        // }
+        // #[cfg(feature = "alloc")]
+        // SignatureScheme::RsaPkcs1Sha384 => {
+        //     use rsa::{
+        //         pkcs1::DecodeRsaPublicKey,
+        //         pkcs1v15::{Signature, VerifyingKey},
+        //         signature::Verifier,
+        //     };
+        //     use sha2::Sha384;
+
+        //     let verifying_key =
+        //         VerifyingKey::<Sha384>::from_pkcs1_der(public_key.try_into().unwrap())
+        //             .map_err(|_| TlsError::DecodeError)?;
+
+        //     let signature =
+        //         Signature::try_from(verify.signature).map_err(|_| TlsError::DecodeError)?;
+        //     verified = verifying_key.verify(message, &signature).is_ok();
+        // }
+        #[cfg(feature = "alloc")]
+        SignatureScheme::RsaPssRsaeSha256 => {
+            use rsa::{
+                RsaPublicKey,
+                pkcs1::DecodeRsaPublicKey,
+                pss::{Signature, VerifyingKey},
+                signature::Verifier,
+            };
+            use sha2::Sha256;
+
+            let der_pubkey = RsaPublicKey::from_pkcs1_der(public_key).unwrap();
+            let verifying_key = VerifyingKey::<Sha256>::from(der_pubkey);
+
+            let signature =
+                Signature::try_from(verify.signature).map_err(|_| TlsError::DecodeError)?;
+            verified = verifying_key.verify(message, &signature).is_ok();
+        }
+        #[cfg(feature = "alloc")]
+        SignatureScheme::RsaPssRsaeSha384 => {
+            use rsa::{
+                RsaPublicKey,
+                pkcs1::DecodeRsaPublicKey,
+                pss::{Signature, VerifyingKey},
+                signature::Verifier,
+            };
+            use sha2::Sha384;
+
+            let der_pubkey = RsaPublicKey::from_pkcs1_der(public_key).unwrap();
+            let verifying_key = VerifyingKey::<Sha384>::from(der_pubkey);
+
+            let signature =
+                Signature::try_from(verify.signature).map_err(|_| TlsError::DecodeError)?;
+            verified = verifying_key.verify(message, &signature).is_ok();
+        }
+        #[cfg(feature = "alloc")]
+        SignatureScheme::RsaPssRsaeSha512 => {
+            use rsa::{
+                RsaPublicKey,
+                pkcs1::DecodeRsaPublicKey,
+                pss::{Signature, VerifyingKey},
+                signature::Verifier,
+            };
+            use sha2::Sha512;
+
+            let der_pubkey = RsaPublicKey::from_pkcs1_der(public_key).unwrap();
+            let verifying_key = VerifyingKey::<Sha512>::from(der_pubkey);
+
+            let signature =
+                Signature::try_from(verify.signature).map_err(|_| TlsError::DecodeError)?;
+            verified = verifying_key.verify(message, &signature).is_ok();
+        }
         _ => {
-            // RSA...
+            error!("InvalidSignatureScheme: {:?}", verify.signature_scheme);
```

Comment:
Maybe "SignatureScheme {:?} is not supported" is more exact than "Invalid signature"